core/foundation/src/FoundationUtils.cxx

```diff
@@ -88,5 +89,89 @@ std::string MakePathRelative(const std::string &path, const std::string &base, b
    }
    return result;
 }
+
+/// Transforms a file path by replacing its backslashes with slashes.
+void ConvertToUnixPath(std::string& Path) {
+   std::replace(Path.begin(), Path.end(), '\\', '/');
+}
+
+const char* GetFallbackRootSys() {
+#ifdef WIN32
+   static char lpFilename[_MAX_PATH];
+   if (::GetModuleFileNameA(
+          NULL,                   // handle to module to find filename for
+          lpFilename,             // pointer to buffer to receive module path
+          sizeof(lpFilename))) {  // size of buffer, in characters
+      auto parent_path = [](std::string& path) {
+         return path.substr(0, path.find_last_of("/\\"));
+      };
+      return StrDup(parent_path(parent_path(lpFilename)).c_str());
+   }
+   return nullptr;
+#else
+   return "/usr/local/root";
+#endif
+}
+
+const std::string& GetRootSys() {
+#ifdef ROOTPREFIX
+   if (IgnorePrefix()) {
+#endif
+      static std::string rootsys;
+      if (rootsys.empty()) {
+         rootsys = ::getenv("ROOTSYS");
+         // We cannot use gSystem->UnixPathName.
+         ConvertToUnixPath(rootsys);
+      }
+      // FIXME: Should this also call UnixPathName for consistency?
+      if (rootsys.empty())
+         rootsys = GetFallbackRootSys();
+      return rootsys;
+#ifdef ROOTPREFIX
+   } else {
+      const static std::string rootsys = ROOTPREFIX;
+      return rootsys;
+   }
+#endif
+}
```

Comment:
If you are going to touch this stuff, you should probably look into fixing [ROOT-9406](https://sft.its.cern.ch/jira/browse/ROOT-9406) by replacing the `#ifdef` with a runtime check (i.e. compute `ROOTSYS` from libdir) + relative dirs (e.g. includedir = ROOTSYS + "/include", etc).

Reply:
I think we can probably do that. We can get the binary location to root.exe at runtime (we do something like that in cling) and we can get the library directory of libCore.so using dladdr I think. I am not sure for the rest -- we do not have a reliable way to find the `etc` dir for instance.

This is probably a good thing to do but outside of the PR I think.

Reply:
This refactoring is necessary to fix the ROOT-10354 which breaks quite a few things with modules.

Reply:
That issue really just looks like a consequence of what's described in ROOT-9406.

Reply:
The issue I am fixing is orthogonal to any installation problems. The issue shows up because we have instructed cling to pick up all modulemap files on the -I path. If we fix all installation issues of ROOT the ROOT-10354 will still exist because we will still find a set of modulemap files which we are not supposed to see.